src/cargo/core/resolver/mod.rs

```diff
@@ -997,42 +1084,54 @@ fn activate_deps_loop(
                 .is_some();
 
         let mut remaining_candidates = RemainingCandidates::new(&candidates);
-        let mut successfully_activated = false;
-        // `conflicting_activations` stores all the reasons we were unable to activate candidates.
-        // One of these reasons will have to go away for backtracking to find a place to restart.
-        // It is also the list of things to explain in the error message if we fail to resolve.
+
+        // `conflicting_activations` stores all the reasons we were unable to
+        // activate candidates. One of these reasons will have to go away for
+        // backtracking to find a place to restart. It is also the list of
+        // things to explain in the error message if we fail to resolve.
+        //
+        // This is a map of package id to a reason why that packaged caused a
+        // conflict for us.
```

Comment:
This is a very well explained description of what is happening. Perhaps it can be used for my nits.